sos/collector/sosnode.py

```diff
@@ -75,6 +77,10 @@ def __init__(self, address, commons, password=None, local_sudo=None,
         self.soslog = logging.getLogger('sos')
         self.ui_log = logging.getLogger('sos_ui')
         self._transport = self._load_remote_transport(commons)
+        # Overwrite need_sudo if transports default_user
+        # is set and is not root.
+        if self._transport.default_user:
+            self.need_sudo = self._transport.default_user != 'root'
```

Comment:
We might split this change and in [sos/collector/transports/__init__.py](https://github.com/sosreport/sos/pull/3192/files#diff-073bf5ff1b0f5ea4cb94d7440664fac5e05e496601ece19838dead00e0912859) into a separate commit, having this change as a generic change & preparation for the juju transport - but I am OK with this.